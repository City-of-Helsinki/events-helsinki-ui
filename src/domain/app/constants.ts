export enum ROUTES {
  ABOUT = '/about',
  COLLECTION = '/collection/:id',
  COLLECTIONS = '/collections',
  EVENTS = '/events',
  EVENT = '/event/:id',
  HOME = '/home',
  HOME_PREVIEW = '/home/:id',
}
