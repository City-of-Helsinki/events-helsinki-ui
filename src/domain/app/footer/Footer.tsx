import React, { FunctionComponent } from 'react';

import BottomFooter from './BottomFooter';
import TopFooter from './TopFooter';

const Footer: FunctionComponent = () => {
  return (
    <footer>
      <TopFooter />
      <BottomFooter />
    </footer>
  );
};

export default Footer;
