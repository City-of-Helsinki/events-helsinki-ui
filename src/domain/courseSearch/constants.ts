export const DEFAULT_SEARCH_FILTERS = {
  categories: [],
  dateTypes: [],
  divisions: [],
  end: null,
  isFree: false,
  onlyOngoingCourses: false,
  places: [],
  start: null,
  text: [],
};
