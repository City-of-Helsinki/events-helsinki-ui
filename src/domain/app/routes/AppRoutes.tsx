import React from 'react';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, Switch } from 'react-router';

import { getCurrentLanguage } from '../../../common/translation/TranslationUtils';
import { SUPPORT_LANGUAGES } from '../../../constants';
import { ROUTES } from './constants';
import App from './LocaleRoutes';

const localeParam = `:locale(${SUPPORT_LANGUAGES.EN}|${SUPPORT_LANGUAGES.FI}|${SUPPORT_LANGUAGES.SV})`;

const AppRoutes: React.FC = () => {
  const { i18n } = useTranslation();
  const locale = getCurrentLanguage(i18n);

  return (
    <Switch>
      {/* Redirect '/' straight to '/:locale/home' to avoid multiple redirections */}
      <Redirect exact path="/" to={`/${locale}${ROUTES.HOME}`} />
      <Redirect exact path={`/${locale}`} to={`/${locale}${ROUTES.HOME}`} />
      <Route path={`/${localeParam}`} component={App} />
      <Route
        render={(props) => (
          // Add locale to path if is is missing
          <Redirect to={`/${locale}${props.location.pathname}`} />
        )}
      />
    </Switch>
  );
};

export default AppRoutes;
