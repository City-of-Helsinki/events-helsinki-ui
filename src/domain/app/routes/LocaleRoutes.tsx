import '../../../assets/styles/main.scss';
import 'hds-core/lib/base.css';
import 'hds-core/lib/components/all.css';

import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, RouteComponentProps, Switch } from 'react-router';

import { SUPPORT_LANGUAGES } from '../../../constants';
import AboutPage from '../../about/AboutPage';
import AccessbilityPage from '../../accessibility/AccessbilityPage';
import CollectionPageContainer from '../../collection/CollectionPageContainer';
import CollectionListPage from '../../collectionList/CollectionListPage';
import CoursePageContainer from '../../course/CoursePageContainer';
import CourseSearchPageContainer from '../../courseSearch/CourseSearchPageContainer';
import EventPageContainer from '../../event/EventPageContainer';
import EventSearchPageContainer from '../../eventSearch/EventSearchPageContainer';
import LandingPage from '../../landingPage/FrontPage';
import LandingPagePreview from '../../landingPage/LandingPagePreview';
import NotFound from '../../notFound/NotFound';
import { ROUTES } from './constants';

const App: FunctionComponent<RouteComponentProps<{
  locale: SUPPORT_LANGUAGES;
}>> = ({
  match: {
    params: { locale },
  },
}) => {
  const { i18n } = useTranslation();

  React.useEffect(() => {
    i18n.changeLanguage(locale);
  }, [i18n, locale]);

  return (
    <Switch>
      <Redirect exact path={`/${locale}/`} to={`/${locale}${ROUTES.HOME}`} />
      <Route exact path={`/${locale}${ROUTES.HOME}`} component={LandingPage} />
      <Route
        exact
        path={`/${locale}${ROUTES.HOME_PREVIEW}`}
        component={LandingPagePreview}
      />
      <Route exact path={`/${locale}${ROUTES.ABOUT}`} component={AboutPage} />
      <Route
        exact
        path={`/${locale}${ROUTES.ACCESSIBILITY}`}
        component={AccessbilityPage}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COLLECTIONS}`}
        component={CollectionListPage}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COLLECTION}`}
        component={CollectionPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENTS}`}
        component={EventSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENT}`}
        component={EventPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.EVENT_PLACE}`}
        component={EventSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COURSES}`}
        component={CourseSearchPageContainer}
      />
      <Route
        exact
        path={`/${locale}${ROUTES.COURSE}`}
        component={CoursePageContainer}
      />
      {/* Redirect to next single event page url */}
      <Redirect
        exact
        path={`/${locale}${ROUTES.EVENT_DEPRECATED}`}
        to={`/${locale}${ROUTES.EVENT}`}
      />
      <Route component={NotFound} />
    </Switch>
  );
};

export default App;
