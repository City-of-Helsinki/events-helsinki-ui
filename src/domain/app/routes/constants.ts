import { MAPPED_PLACES } from '../../eventSearch/constants';

// Remember also update static urls to updateSitemap
export type ROUTE_TYPES = {
  ABOUT: '/about';
  ACCESSIBILITY: '/accessibility';
  COLLECTION: '/collection/:slug';
  COLLECTIONS: '/collections';
  EVENTS: '/events';
  EVENT: '/event/:id';
  COURSES: '/courses';
  COURSE: '/course/:id';
  HOME: '/home';
  HOME_PREVIEW: '/home/:id';
  EVENT_PLACE: string;
};

export const ROUTES: ROUTE_TYPES = {
  ABOUT: '/about',
  ACCESSIBILITY: '/accessibility',
  COLLECTION: '/collection/:slug',
  COLLECTIONS: '/collections',
  EVENTS: '/events',
  EVENT: '/event/:id',
  COURSES: '/courses',
  COURSE: '/course/:id',
  HOME: '/home',
  HOME_PREVIEW: '/home/:id',
  EVENT_PLACE: `/:place(${Object.keys(MAPPED_PLACES).join('|')})`,
};
