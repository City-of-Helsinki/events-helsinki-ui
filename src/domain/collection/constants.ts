import { mockEventData } from "../event/constants";

export const mockCollection = {
  collectionDetails: {
    curatedEvents: [
      `http://localhost:3000/fi/event/${mockEventData.eventDetails.id}`
    ],
    curatedEventsTitle: {
      en: "At least visit these",
      fi: "Käy ainakin näissä",
      sv: "Besök åtminstone dessa"
    },
    description: {
      en:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      fi:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      sv:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus."
    },
    eventListQuery:
      "http://localhost:3000/en/events?categories=music&districts=kaupunginosa%3Aetu-t%C3%B6%C3%B6l%C3%B6",
    eventListTitle: {
      en: "All the best events of the fall",
      fi: "Kaikki syksyn parhaat tapahtumat",
      sv: "Höstens bästa händelser"
    },
    id: "1",
    linkText: {
      en: "Read more on the project website",
      fi: "Lue lisää hankkeen omilta sivuilta",
      sv: "Läs mer på projektets webbplats"
    },
    linkUrl: {
      en: "http://www.google.com",
      fi: "http://www.google.com",
      sv: "http://www.google.com"
    },
    similarCollectionsTitle: {
      en: "Check also these",
      fi: "Katso myös nämä",
      sv: "Se dessa också"
    },
    socialMediaDescription: {
      en: "We put together the best foliage for the fall",
      fi: "Kokosimme parhaat tärpit syksylle",
      sv: "Vi sätter ihop det bästa bladverket för hösten"
    },
    subtitles: {
      en: "Subtitles en",
      fi: "Subtitles fi",
      sv: "Subtitles sv"
    },
    title: {
      en: "The most bustling tricks of the fall",
      fi: "Syksyn riehakkaimmat riennot",
      sv: "Höstens mest livliga knep"
    }
  }
};

// Amount of the similar collections
export const SIMILAR_COLLECTIONS_AMOUNT = 8;
