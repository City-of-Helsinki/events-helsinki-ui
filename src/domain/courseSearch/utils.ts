import { forEach, isArray, isEmpty, isNil, isNumber } from 'lodash';

import { Meta } from '../../generated/graphql';
import { formatDate } from '../../util/dateUtils';
import getUrlParamAsArray from '../../util/getUrlParamAsArray';
import { EVENT_SEARCH_FILTERS } from '../eventSearch/constants';
import { Filters, MappedFilters } from './types';

export const getSearchFilters = (searchParams: URLSearchParams) => {
  const endTime = searchParams.get(EVENT_SEARCH_FILTERS.END);
  const end = endTime ? new Date(endTime) : null;

  const startTime = searchParams.get(EVENT_SEARCH_FILTERS.START);
  const start = startTime ? new Date(startTime) : null;

  return {
    categories: getUrlParamAsArray(
      searchParams,
      EVENT_SEARCH_FILTERS.CATEGORIES
    ),
    dateTypes: getUrlParamAsArray(
      searchParams,
      EVENT_SEARCH_FILTERS.DATE_TYPES
    ),
    divisions: getUrlParamAsArray(searchParams, EVENT_SEARCH_FILTERS.DIVISIONS),
    places: getUrlParamAsArray(searchParams, EVENT_SEARCH_FILTERS.PLACES),
    text: getUrlParamAsArray(searchParams, EVENT_SEARCH_FILTERS.TEXT),
    start,
    end,
    alsoOngoingCourses:
      searchParams.get(EVENT_SEARCH_FILTERS.ALSO_ONGOING_COURSES) === 'true'
        ? true
        : false,
  };
};

export const getSearchQuery = (filters: Filters): string => {
  const newFilters: MappedFilters = {
    ...filters,
    start: formatDate(filters.start, 'yyyy-MM-dd'),
    end: formatDate(filters.end, 'yyyy-MM-dd'),
    alsoOngoingCourses: filters.alsoOngoingCourses ? true : undefined,
  };

  if (newFilters.end || newFilters.start) {
    delete newFilters.dateTypes;
  }

  const query: string[] = [];

  forEach(newFilters, (filter, key) => {
    if (!isEmpty(filter) || isNumber(filter) || typeof filter === 'boolean') {
      if (isArray(filter)) {
        const items: Array<string | number> = [];

        forEach(filter, (item: string | number) => {
          items.push(encodeURIComponent(item));
        });

        query.push(`${key}=${items.join(',')}`);
      } else if (!isNil(filter)) {
        query.push(`${key}=${encodeURIComponent(filter)}`);
      }
    }
  });

  return query.length ? `?${query.join('&')}` : '';
};

export const getNextPage = (meta: Meta): number | null => {
  if (!meta.next) return null;

  const urlParts = meta.next.split('?');
  const searchParams = new URLSearchParams(decodeURIComponent(urlParts[1]));
  const page = searchParams.get(EVENT_SEARCH_FILTERS.PAGE);
  return page ? Number(page) : null;
};
