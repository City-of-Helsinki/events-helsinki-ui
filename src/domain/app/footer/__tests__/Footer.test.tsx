import * as React from 'react';

import { setFeatureFlags } from '../../../../test/feature-flags/featureFlags.test.utils';
import { render, screen, userEvent } from '../../../../test/testUtils';
import { isFeatureEnabled } from '../../../../util/featureFlags';
import { skipFalsyType } from '../../../../util/typescript.utils';
import { ROUTES } from '../../routes/constants';
import Footer from '../Footer';

[true, false].forEach((EVENTS_HELSINKI_2) => {
  describe('EVENTS_HELSINKI_2 feature flag', () => {
    beforeEach(() => {
      setFeatureFlags({ EVENTS_HELSINKI_2 });
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    it(`matches snapshot when EVENTS_HELSINKI_2 feature is ${
      EVENTS_HELSINKI_2 ? 'on' : 'off'
    }`, () => {
      const { container } = render(<Footer />);

      expect(container.firstChild).toMatchSnapshot();
    });

    it(`should route to right place when clicking links and EVENTS_HELSINKI_2 feature is ${
      EVENTS_HELSINKI_2 ? 'on' : 'off'
    }`, async () => {
      const { history } = render(<Footer />);
      const pushSpy = jest.spyOn(history, 'push');

      const links = [
        {
          linkName: 'Tapahtumat',
          path: `/fi${ROUTES.EVENTS}`,
        },
        isFeatureEnabled('EVENTS_HELSINKI_2') && {
          linkName: 'Harrastukset',
          path: `/fi${ROUTES.COURSES}`,
        },
        {
          linkName: 'Suosittelemme',
          path: `/fi${ROUTES.COLLECTIONS}`,
        },
      ].filter(skipFalsyType);
      expect(links).toHaveLength(EVENTS_HELSINKI_2 ? 3 : 2);
      for (const { linkName, path } of links) {
        userEvent.click(screen.getByRole('link', { name: linkName }));
        expect(pushSpy).toHaveBeenCalledWith(path);
        pushSpy.mockReset();
      }
    });
  });
});
test('should show courses footer title', () => {
  render(<Footer />, {
    routes: [`/fi/courses`],
  });
  expect(
    screen.queryByText(/suositut harrastuskategoriat/i)
  ).toBeInTheDocument();
});

test('should show events footer title', () => {
  render(<Footer />, {
    routes: [`/fi/events`],
  });
  expect(
    screen.queryByText(/suositut tapahtumien kategoriat/i)
  ).toBeInTheDocument();
});

test('should not show footer title', () => {
  render(<Footer />, {
    routes: [`/fi/home`],
  });
  expect(
    screen.queryByText(/suositut harrastuskategoriat/i)
  ).not.toBeInTheDocument();
  expect(
    screen.queryByText(/suositut tapahtumien kategoriat/i)
  ).not.toBeInTheDocument();
});
