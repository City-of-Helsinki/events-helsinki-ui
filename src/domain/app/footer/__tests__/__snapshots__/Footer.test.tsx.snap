// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer>
  <footer
    className="topFooterWrapper"
  >
    <div
      className="container"
    >
      <div
        className="companyInfoWrapper"
      >
        <div
          className="logoWrapper"
        >
          <div
            className="helsinkiLogo"
          />
          <h2
            className="appName"
          >
            Tapahtumat
          </h2>
        </div>
        <div
          className="iconLinkWrapper"
        >
          <div
            className="iconLink"
          >
            <a
              href="/fi/events"
              onClick={[Function]}
            >
              <div
                className="iconWrapper"
              >
                <svg
                  className=""
                  height="100%"
                  viewBox="0 0 24 24"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.3582734,0.0535251799 C12.0567815,0.0517959979 9.04784082,1.94662012 7.62269829,4.92467659 C6.19755575,7.90273306 6.6096026,11.4345925 8.68201439,14.0046043 L0.00920863309,22.676259 L1.30820144,23.9758273 L9.98273381,15.3035971 C12.9168981,17.6690044 17.0515408,17.841428 20.1724602,15.7285309 C23.2933795,13.6156338 24.6689728,9.71271903 23.5625765,6.10989254 C22.4561801,2.50706606 19.1271543,0.0488985453 15.3582734,0.0517959979 L15.3582734,0.0535251799 Z M15.3582734,15.3657554 C11.637384,15.3657554 8.62100719,12.3493786 8.62100719,8.62848921 C8.62100719,4.90759984 11.637384,1.89122302 15.3582734,1.89122302 C19.0791628,1.89122302 22.0955396,4.90759984 22.0955396,8.62848921 C22.0917339,12.3479875 19.0777709,15.3624653 15.3582734,15.3669065 L15.3582734,15.3657554 Z"
                  />
                </svg>
              </div>
              <div
                className="textWrapper"
              >
                Etsi tekemist√§
              </div>
            </a>
          </div>
          <div
            className="iconLink"
          >
            <a
              href="/fi/collections"
              onClick={[Function]}
            >
              <div
                className="iconWrapper"
              >
                <svg
                  className=""
                  height="100%"
                  viewBox="0 0 24 24"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.013717,1.01899281 C18.0697842,1.02004796 22.9794245,5.92968825 22.9794245,11.9852278 C22.9794245,18.0407674 18.0682014,22.9504077 12.013717,22.9504077 C5.95923261,22.9504077 1.04853717,18.041295 1.04853717,11.9852278 C1.04853717,5.92916067 5.95923261,1.02004796 12.013717,1.01899281 L12.013717,1.01899281 Z M12.013717,2.84652278 C6.97534772,2.84652278 2.87764988,6.94685851 2.87764988,11.9852278 C2.87764988,17.0235971 6.97640288,21.1228777 12.0152998,21.1228777 C17.0541966,21.1228777 21.1534772,17.0235971 21.1534772,11.9852278 C21.1534772,6.94685851 17.0526139,2.84652278 12.013717,2.84652278 Z M16.3688729,14.1029257 L18.0117506,14.903789 C16.9025481,17.205879 14.5690751,18.6656081 12.013717,18.6559233 C9.45683429,18.6671944 7.12141723,17.2069429 6.01199041,14.9032614 L7.65486811,14.1029257 C8.46605189,15.7702798 10.157662,16.828478 12.0118705,16.828478 C13.866079,16.828478 15.5576891,15.7702798 16.3688729,14.1029257 Z M8.0542446,8.33016787 C8.89543918,8.33016787 9.57736211,9.0120908 9.57736211,9.85328537 C9.57736211,10.6944799 8.89543918,11.3764029 8.0542446,11.3764029 C7.21305003,11.3764029 6.5311271,10.6944799 6.5311271,9.85328537 C6.5311271,9.0120908 7.21305003,8.33016787 8.0542446,8.33016787 Z M15.973717,8.33016787 C16.8149116,8.33016787 17.4968345,9.0120908 17.4968345,9.85328537 C17.4968345,10.6944799 16.8149116,11.3764029 15.973717,11.3764029 C15.1325225,11.3764029 14.4505995,10.6944799 14.4505995,9.85328537 C14.4505995,9.0120908 15.1325225,8.33016787 15.973717,8.33016787 Z"
                  />
                </svg>
              </div>
              <div
                className="textWrapper"
              >
                Selaa kokoelmia
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="categoriesWrapper"
      >
        <h2
          className="categoriesTitle"
        >
          Kategoriat
        </h2>
        <div
          className="categoryFilters"
        >
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                movie.svg
              </svg>
              <span>
                Elokuva
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                music.svg
              </svg>
              <span>
                Musiikki
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                sport.svg
              </svg>
              <span>
                Liikunta ja urheilu
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                museum.svg
              </svg>
              <span>
                Museot
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                dance.svg
              </svg>
              <span>
                Tanssi
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                culture.svg
              </svg>
              <span>
                Kulttuuri ja taide
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg
                className=""
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.250564,11.2321141 C17.250564,10.8655334 17.497019,10.5435063 17.896874,10.0494683 C18.3886561,9.44432807 19,8.69086368 19,7.56461489 C19,5.62455688 17.3537705,4.8519175 16.469465,4.43627135 C16.2686916,4.3420883 16.0414115,4.23606188 15.9624557,4.17346117 C15.7690139,4.02288108 15.5626007,3.68111505 15.3437802,3.32017403 C14.7854496,2.39582662 13.940622,1 12,1 C10.0283597,1 9.21737029,2.40767 8.68159845,3.33765711 C8.4751853,3.69521431 8.2811795,4.03190461 8.09788914,4.1785369 C8.0206252,4.24057364 7.77247825,4.35505962 7.57339671,4.44755076 C6.673864,4.86263294 5,5.63640026 5,7.56461489 C5,8.69086368 5.61134386,9.44432807 6.10256204,10.0494683 C6.50354496,10.5435063 6.74887206,10.8655334 6.74887206,11.2321141 C6.74887206,11.5090235 6.66089268,11.6455044 6.44150822,11.9539961 C6.15162746,12.3640026 5.75402836,12.9240251 5.75402836,13.9662423 C5.75402836,15.6564615 7.1741057,16.7088302 8.54737351,17.7279246 C9.93699646,18.7577345 11.2493555,19.7305833 11.2493555,21.2645827 L11.2493555,21.2668385 L11.2493555,22.7540284 L12.7506445,22.7540284 L12.7506445,21.2645827 C12.7506445,19.7305833 14.0630035,18.7577345 15.4526265,17.7279246 C16.8253303,16.7088302 18.2459716,15.6564615 18.2459716,13.9662423 C18.2459716,12.9240251 17.8489365,12.3640026 17.5579278,11.9539961 C17.3391073,11.6455044 17.250564,11.5090235 17.250564,11.2321141 M16.3341121,12.8213825 C16.5726716,13.1586368 16.7452465,13.402272 16.7452465,13.9662423 C16.7452465,14.9013052 15.7323558,15.6519497 14.5592975,16.5227199 C13.9479536,16.9755881 13.3050274,17.4521431 12.7506445,18.0042701 L12.7506445,16.0771834 L15.0347245,13.7919755 L13.9738962,12.7311473 L12.7506445,13.9549629 L12.7506445,11.0764583 L15.5304544,8.29552046 L14.4696262,7.23412826 L12.7506445,8.95423783 L12.7506445,5.26417983 L11.2493555,5.26417983 L11.2493555,8.95423783 L9.53037383,7.23412826 L8.4695456,8.29552046 L11.2493555,11.0764583 L11.2493555,13.9549629 L10.0255398,12.7311473 L8.96527554,13.7919755 L11.2493555,16.0771834 L11.2493555,18.0042701 C10.6944086,17.4521431 10.0520464,16.9755881 9.44126652,16.5227199 C8.26764422,15.6519497 7.25418949,14.9013052 7.25418949,13.9662423 C7.25418949,13.402272 7.42676442,13.1586368 7.66532388,12.8213825 C7.92587818,12.4548018 8.24903319,11.9979858 8.24903319,11.2321141 C8.24903319,10.3139704 7.72792459,9.67160812 7.26772478,9.10481792 C6.85602643,8.59668063 6.50016113,8.15903964 6.50016113,7.56461489 C6.50016113,7.01530777 6.62592652,6.53818885 8.20278762,5.80897519 C8.53158234,5.65670319 8.8158234,5.52586207 9.03520786,5.34990332 C9.43901063,5.02674831 9.70182082,4.56993232 9.98042217,4.0871737 C10.5494683,3.09966162 10.9696262,2.5007251 12,2.5007251 C13.0213503,2.5007251 13.4595553,3.10304544 14.0601837,4.0967612 C14.3523203,4.57839188 14.6286658,5.03520786 15.0381083,5.35497905 C15.2507251,5.52135031 15.5203029,5.64824364 15.8316146,5.79431196 C16.8078472,6.25338382 17.4992749,6.64816307 17.4992749,7.56461489 C17.4992749,8.15903964 17.1439736,8.59668063 16.7317112,9.10481792 C16.2726394,9.67160812 15.7504028,10.3139704 15.7504028,11.2321141 C15.7504028,11.9979858 16.0741218,12.4548018 16.3341121,12.8213825"
                />
              </svg>
              <span>
                Luonto ja ulkoilu
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg
                className=""
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.8294615,20.5124154 C16.4065385,20.5124154 17.2068462,20.6979538 17.9808462,20.8765692 C18.6911538,21.0420308 19.362,21.1971077 20.7417692,21.1971077 L20.7417692,22.9050308 C19.1681538,22.9050308 18.3678462,22.7201846 17.5945385,22.5422615 C16.8835385,22.3761077 16.2113077,22.2210308 14.8294615,22.2210308 C13.4476154,22.2210308 12.7760769,22.3761077 12.0643846,22.5415692 C11.2917692,22.7201846 10.4914615,22.9050308 8.91507692,22.9050308 C7.34007692,22.9050308 6.53907692,22.7201846 5.76507692,22.5415692 C5.05338462,22.3761077 4.38115385,22.2210308 3,22.2210308 L3,20.5124154 C4.57638462,20.5124154 5.37669231,20.6979538 6.15,20.8765692 C6.86169231,21.0420308 7.53392308,21.1971077 8.91507692,21.1971077 C10.2962308,21.1971077 10.9677692,21.0420308 11.6787692,20.8765692 C12.4527692,20.6979538 13.2530769,20.5124154 14.8294615,20.5124154 Z M3.90830769,14.7241 L6.39023077,17.2053308 L3,18.1136385 L3.90830769,14.7241 Z M18.9021692,3.86234615 L20.1164769,5.07665385 L8.57293846,16.6208846 L7.35863077,15.4065769 L18.9021692,3.86234615 Z M16.0396154,1 L17.2546154,2.215 L5.71038462,13.7585385 L4.49607692,12.5442308 L16.0396154,1 Z"
                />
              </svg>
              <span>
                Osallistu ja vaikuta
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg>
                theatre.svg
              </svg>
              <span>
                Teatteri
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="categoryFilter withHorizontalPadding"
          >
            <button
              onClick={[Function]}
            >
              <svg
                className=""
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.9714253,10.9514872 C14.1495802,10.7554905 14.4021203,10.6437617 14.6669574,10.6437617 L15.7581797,10.6437617 C16.2917391,10.6437617 16.8001765,10.4170603 17.1567126,10.0201253 L21,5.74180501 L20.4183844,5.16018943 L17.5935184,7.98505543 L17.1781003,7.56963729 L19.9198676,4.66167257 L19.3382897,4.08009472 L16.4303627,6.821862 L16.0149068,6.40648158 L18.8397728,3.58161558 L18.258195,3 L13.9798747,6.84324964 C13.5829397,7.19982348 13.3562006,7.70826089 13.3562006,8.24182028 L13.3562006,9.4018829 C13.3562006,9.66513571 13.2458298,9.91628017 13.0519453,10.0943219 L11.7379445,11.309155 L3.64251505,3.18784912 C3.39091794,2.93621428 2.96044924,3.14269743 3.00292275,3.49606501 C3.24086497,5.47538366 4.30553191,7.76171113 6.54153997,9.99771918 L8.18653998,11.6427569 L9.95854562,11.6427569 L10.5907069,12.3698613 L4.03439548,18.4314657 L5.65114268,20.0338036 L11.6041869,13.5355065 L17.3328306,20.1243333 L18.9900521,18.5844608 L12.7310553,12.305472 L13.9714253,10.9514872 Z"
                />
              </svg>
              <span>
                Ruoka
              </span>
              <svg
                className=""
                height="100%"
                viewBox="0 0 199.404 199.404"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                  />
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </footer>
  <div
    className="bottomFooter"
  >
    <div
      className="container"
    >
      <div
        className="contentWrapper"
      >
        <div
          className="linkContainer"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            Tietoa palvelusta
          </a>
          <a
            href="/faq"
            onClick={[Function]}
          >
            Usein kysytty√§
          </a>
          <a
            href="/feedback"
            onClick={[Function]}
          >
            Anna palautetta
          </a>
        </div>
        <div
          className="logoWrapper"
        >
          <div
            className="helsinkiLogo"
          />
        </div>
      </div>
    </div>
  </div>
</footer>
`;
